x-pack/plugins/security_solution/public/common/components/exceptions/helpers.tsx

```diff
@@ -397,6 +381,107 @@ export const entryHasListType = (
   return false;
 };
 
+/**
+ * Returns the value for `file.Ext.code_signature` which
+ * can be an object or array of objects
+ */
+export const getCodeSignatureValue = (
+  alertData: Ecs
+): Array<{ subjectName: string; trusted: string }> => {
+  const { file } = alertData;
+  const codeSignature = file && file.Ext && file.Ext.code_signature;
+
+  // Pre 7.10 file.Ext.code_signature was mistakenly populated as
+  // a single object with subject_name and trusted.
+  if (Array.isArray(codeSignature) && codeSignature.length > 0) {
+    return codeSignature.map((signature) => ({
+      subjectName: (signature.subject_name && signature.subject_name[0]) ?? '',
+      trusted: (signature.trusted && signature.trusted[0]) ?? '',
+    }));
+  } else {
+    const codeSignatures: CodeSignature | CodeSignature[] | undefined =
+      file && file.Ext && file.Ext.code_signature;
```

Comment:
Why do we need to define a `codeSignatures` variable here? Isn't it the same the same as the `codeSignature` variable defined above? https://github.com/elastic/kibana/pull/77611/files#diff-6d42f8fa8de87130b16d9c8b04c428afR392

Reply:
Good catch! Thought I had to do it to get some of the typings read right (to avoid them turning to any), but removed and still looks good. Thanks!